packages/components/src/components/date-picker/date-picker.tsx

```diff
@@ -36,22 +74,22 @@ const DatePicker = React.memo(props => {
         ...other_props
     } = props;
 
-    const datepicker_ref = React.useRef();
-    const calendar_ref = React.useRef();
-    const calendar_el_ref = React.useRef();
-    const [placement, setPlacement] = React.useState('');
-    const [style, setStyle] = React.useState({});
-    const [date, setDate] = React.useState(value ? toMoment(value).format(display_format) : '');
-    const [duration, setDuration] = React.useState(daysFromTodayTo(value));
-    const [is_datepicker_visible, setIsDatepickerVisible] = React.useState(false);
-    const [is_placeholder_visible, setIsPlaceholderVisible] = React.useState(placeholder && !value);
+    const datepicker_ref = React.useRef<HTMLDivElement | null>(null);
+    const calendar_ref = React.useRef<HTMLElement | null>(null);
```

Comment:
```suggestion
    const calendar_ref = React.useRef<HTMLElement>(null);
```
useRef built-in type already assumes that it can be null if you pass `null` as initial value :)